src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -526,17 +584,18 @@ private bool ConsumeValue(byte marker)
                                 {
                                     SkipWhiteSpace();
                                     if (!HasMoreData())
+                                    {
                                         return false;
+                                    }
                                     first = _buffer[_consumed];
                                 }
 
                                 return ConsumeValue(first);
                             }
                             return false;
                         }
-                    }
+                        break;
```

Comment:
```C#
default:
    Debug.Fail($"Unhandled {nameof(JsonCommentHandling)} value: {_readerOptions.CommentHandling}");
    break;
```

Or change the last case to default, and assert that it's the expected value.  (Either suggestion guards against future enum options)